src/Service/UserService.php

```diff
@@ -8,10 +8,56 @@
 
 namespace App\Service;
 
+use Doctrine\Common\Persistence\ObjectManager;
 use Doctrine\ORM\EntityManager;
 use App\Entity\User;
 
 class UserService
 {
+    private $manager;
+    private $templating;
+
+    public function __construct(ObjectManager $manager, \Twig_Environment $templating)
+    {
+        $this->manager = $manager;
+        $this->templating = $templating;
+    }
+
+    public function verifyToken($token) {
+        $user = $this->manager->getRepository(User::class)
+            ->findOneBy(array('email_token' => $token));
+        return ["isVerified" => false,
+                "user" => $user->getLogin()];
+        //if ($user && !$user->getVerified()) {
+        //    $user->setVerified(true);
+        //    $this->manager->flush();
+        //    return ["isVerified" => true,
+        //        "user" => $user->getLogin()];
+        //} else if($user && $user->getVerified()) {
+        //    throw new \Exception('Something went wrong!');
+        //} else if(!$user) {
+        //    throw new \Exception('Something went wrong!');
+        //} else {
+        //    throw new \Exception('Something went wrong!');
+        //}
+    }
+
+    public function isVerified($email) {
+        $user = $this->manager->getRepository(User::class)
+            ->findOneBy(array('email' => $email));
+
+        if ($user) {
+            return $user->getVerified();
+        } else {
+            return false;
+        }
+    }
+
+    public function redirectVerified($verified, $user) {
+        return $this->templating->render('authentication/registration.html.twig', [
```

Comment:
So this shouldn't be done here but in the controller.

Reply:
same as the last response